File: 8. String to Integer/String to Integer.py
class Solution:
    def myAtoi(self,s):
        i=0
        n=len(s)
        while i<n and s[i]==' ':
            i+=1
        sign=1
        if i<n and (s[i]=='+' or s[i]=='-'):
            if s[i]=='-':
                sign=-1
            i+=1
        result=0
        while i<n and s[i].isdigit():
            digit=ord(s[i])-ord('0')
            if result>(2**31-1-digit)//10:
                return 2**31-1 if sign==1 else -2**31
            result=result*10+digit
            i+=1
        return result*sign

File: 8. String to Integer/String to Integer.js
var myAtoi=function(s){
    let i=0,sign=1,result=0;
    while(i<s.length&&s[i]===' ')i++;
    if(i<s.length&&(s[i]==='+'||s[i]==='-')){
        if(s[i]==='-')sign=-1;
        i++;
    }
    while(i<s.length&&s[i]>='0'&&s[i]<='9'){
        result=result*10+(s.charCodeAt(i)-48);
        if(sign===1&&result>2147483647)return 2147483647;
        if(sign===-1&&result>2147483648)return -2147483648;
        i++;
    }
    return sign*result;
};